apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -848,6 +848,20 @@ async function legacyTraverseQuestionAndExecuteFunctions(phase, codeCaller, data
   };
 }
 
+/**
+ * @param {unknown} val
+ * @return {{ score: number, weight: number }}
```

Comment:
The PartialScore dict in python also has a feedback property -- can we make a proper PartialScore type in TS and use that instead?

Reply:
That Python type is at best aspirational. It doesn't reflect what's actually going to be there, and at any rate, `feedback` isn't used anywhere here.

Reply:
Now that I think about it, what is the difference between the `feedback` dict and the `partial_scores` feedback field?

Reply:
`feedback` is essentially general-purpose scratch space for the question as a whole and is mostly used by `server.py` and `question.html`. Elements will never read from or write to it (with the exception of `<pl-external-grader-results>`, which is a bit of a special case: that'll read feedback). `partial_scores.*.feedback` is how elements track feedback associated with specific "answer names" (read: inputs).

Reply:
Got it, that clears that up.